Readme updates to support ongoing development (#2406)
geeksforsocialchange/PlaceCal: README.md

# PlaceCal

## Introduction

> PlaceCal is an online calendar which lists events and activities by and for members of local communities, curated around interests and locality.

Reviewer: Is there any wording we can think of to say that it is also a directory? Or something like that.

Reviewer 2: i'm in the process of commissioning sarah for some new hero copy so we can change it then!

> The codebase doesn't currently have enough seeds to create a dev environment suitable for developing from scratch. If you're interested in contributing to PlaceCal please get in touch with us at support@Placecal.org

Reviewer 2: lowercase P in the email too pls :)

To get an idea of the project and what we're about, check out [the handbook](https://handbook.placecal.org/).

## Requirements

To run PlaceCal locally you will need to install the following dependencies:

> - Docker

Reviewer: ```suggestion
- [Docker](https://docs.docker.com/get-docker/)
```

Reviewer: Oh I see you removed the links deliberately? Fine either way.

Author: I'll re-list them all as links, doesn't take up any more space or reading and if it might be useful to someone then it makes sense to include

Some other useful commands for developing can be found in the makefile.

## Troubleshooting

If the make command fails and you can't work out why, here are some suggestions:

> - Do you have an older version of the database hanging around? Try running `rails:db:drop:all`

Reviewer 2: ```suggestion
- Do you have an older version of the database hanging around? Try running `rails db:drop:all`
```

- The admin interface is at `admin.lvh.me:3000`
- Access code docs through your local filesystem and update them with `bin/rails yard`

## Testing, linting and formatting

PlaceCal tests are written in minitest. We use Rubocop to lint our Ruby code.

We use [Prettier](https://prettier.io/) to format everything it's able to parse. We run this automatically as part of a pre-commit hook.

You can run the tests & rubocop with:

```sh
make test
```

## Documentation for Developers

> The documentation for PlaceCal currently stored in notion and can be read [here](https://www.notion.so/gfsc/PlaceCal-developer-handbook-01649b69009340e3ae3035e9cf346f27). There is also a small amount of documentation sprinkled throughout the code itself and can be turned into HTML by running `rails yard`. If you are working with the code and are completely lost you can also try the [GFSC discord server](http://discord.gfsc.studio) where you can prod a human for answers. Good Luck!

Reviewer: ```suggestion
The documentation for PlaceCal is currently stored in notion and can be read [here](https://www.notion.so/gfsc/PlaceCal-developer-handbook-01649b69009340e3ae3035e9cf346f27). There is also a small amount of documentation sprinkled throughout the code itself and can be turned into HTML by running `rails yard`. If you are working with the code and are completely lost you can also try the [GFSC discord server](http://discord.gfsc.studio) where you can prod a human for answers. Good Luck!
```

> ### Generating new components

Reviewer 2: Should we add a note here about the legacy mountain_view system? I think this could be really confusing for new people otherwise but understand if you want to keep the README ultra simple.

Author: sure, have added a line

We use view_component to make components, and you can create a new one by running `rails g component <Name> <args>`
More info here: https://viewcomponent.org/guide/generators.html

## Donations

> If you'd like to support development, please consider sending us a one-off or regular donation on Ko-fi.

Reviewer 2: ```suggestion
If you'd like to support development, please consider sending us a one-off or regular donation. You can do this through the "support" button in GitHub at the top of this repo.
```

## Documentation for Developers

The documentation for PlaceCal is currently stored in notion and can be read [here](https://handbook.placecal.org/placecal-developer-handbook). There is also a small amount of documentation sprinkled throughout the code itself and can be turned into HTML by running `rails yard`. If you are working with the code and are completely lost you can also try the [GFSC discord server](http://discord.gfsc.studio) where you can prod a human for answers. Good Luck!

### Generating new components

We use view_component to make components, and you can create a new one by running `rails g component <Name> <args>`
Previously this system used mountain view, and some of the components are still generated using this.
More info here: https://viewcomponent.org/guide/generators.html

## Donations

> If you'd like to support development, please consider sending us a one-off or regular donation on Ko-fi. You can do this through the "support" button in GitHub at the top of this repo.

Reviewer 2: ```suggestion
If you'd like to support development, please consider sending us a one-off or regular donation. You can do this through the "support" button in GitHub at the top of this repo.
```
